pkg/csv/ingest_test.go

```diff
@@ -0,0 +1,101 @@
+package csv
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/bserdar/digraph"
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIngest(t *testing.T) {
+	schStr := `{
+		"@context": ["../../schemas/ls.json"],
+		"@type": "Schema",
+		"@id": "http://example.com/id",
+		"layer": {
+			"@type": "Object",
+			"attributeList": [
+				{
+					"@id": "https://www.example.com/id1",
+					"@type": "Value",
+					"attributeName": "field1"
+				},
+				{
+					"@id": "https://www.example.com/id2",
+					"@type": "Value",
+					"attributeName": "field2"
+				},
+				{
+					"@id": "https://www.example.com/id3",
+					"@type": "Value",
+					"attributeName": "field3"
+				},
+				{
+					"@id": "https://www.example.com/id4",
+					"@type": "Value",
+					"attributeName": "field4"
+				},
+				{
+					"@id": "https://www.example.com/id5",
+					"@type": "Value",
+					"attributeName": "field5"
+				},
+				{
+					"@id": "https://www.example.com/id6",
+					"@type": "Value",
+					"attributeName": "field6"
+				}
+			]
+		}
+	}
+	`
+
+	inputStrColData := [][]string{
+		{"data1", "data2", "data3", "data4", "data5", "data6"},
+		{"data1", "data2", "data3", "data4", "data5", "data6", "data7"},
+		{"data1", "data2", "data3", "data4", "data5", "data6", "data7", "data8", "data9"},
+		{"data1", "data2", "data3", "data4", "data5"},
+		{"data1", "data2", "data3"},
+	}
+
```

Comment:
Create an expectedNodes[][] for each input row, and check if the ingested data has all the expected nodes, and nothing more.


Then write a new test with the OnlySchemaNodes flag set and make sure only schema nodes are imported